### ISSUE TITLE
macos: tweak fsevents event delivery

### ISSUE THREAD

**Issue by bnoordhuis** [libuv/libuv#3866]
Pointed out by @vtjnash in https://github.com/libuv/libuv/pull/3863#discussion_r1053424899:

1. Libuv calls `FSEventStreamCreate()` with `latency = 0.05` (50 ms) - affects event coalescing. Not sure if it should be bigger or smaller.

2. Libuv discards events from before the `uv_fs_event_init()` call. Maybe it shouldn't.

3. Libuv doesn't call `FSEventStreamFlushSync` (or the async version.) Maybe it should.

Hopefully with the right tweaks event delivery gets more deterministic.

**Comment by bharsaklemukesh975** [comment]
Can I work on this?

**Comment by bnoordhuis** [comment]
@bharsaklemukesh975 sure, go for it

**Comment by bharsaklemukesh975** [comment]
@bnoordhuis Thanks, I will work on fsevents event delivery.

**Comment by bharsaklemukesh975** [comment]
Changes I should do: 

- Libuv should not discards events from before the uv_fs_event_init() call.
- Libuv should call call FSEventStreamFlushSync

right?

**Comment by bnoordhuis** [comment]
@bharsaklemukesh975 I'd rephrase that as: think about whether it makes sense to make those changes, because the changes themselves are fairly trivial. The important part is to think through the consequences and ramifications.

To take (2) as an example: should libuv discard those events? If yes, why? If not, why not? How are existing programs affected?

**Comment by juanarbol** [comment]
I’ll adjust some knobs and see how that works.
